sources/packages/api/src/services/student/student.service.ts

```diff
@@ -50,6 +50,40 @@ export class StudentService extends RecordDataModelService<Student> {
     return student;
   }
 
+  /**
+   * Gets the student by his id.
+   * @param studentId student id.
+   * @returns the student found or null.
+   */
+  async getStudentById(studentId: number): Promise<Student> {
+    return this.repo
+      .createQueryBuilder("student")
+      .select([
+        "student.id",
+        "student.birthDate",
+        "user.id",
+        "user.firstName",
+        "user.lastName",
+        "user.email",
+      ])
+      .innerJoin("student.user", "user")
+      .where("student.id = :studentId", { studentId })
+      .getOne();
+  }
+
+  /**
+   * Check if the student id exists on DB.
+   * @param studentId student to be verified.
+   * @returns true if the student was found, otherwise false.
+   */
+  async studentExists(studentId: number): Promise<boolean> {
+    const studentFound = this.repo.findOne({
```

Comment:
why didn't we use a select 1 here like we did for other cases. Both are apparently the same. just asking.

Reply:
The parameters expect a property from the `Student` object.